### ISSUE TITLE
blobsaver becomes unresponsive

### ISSUE THREAD

**Issue by skiasaurus** [airsquared/blobsaver#136]
**Describe the bug**

I've been running blobsaver on Windows 10. I updated to 2.4.1 when prompted (without uninstalling 2.4.0 first), and now when I click "go" in the window it (technically Java SE platform) immediately becomes unresponsive.

I first noticed this issue when I restarted my computer last night. When I logged in and blobsaver started automatically (as I've had it set to do), it prompted me to update to 2.4.1, even though I'd already done so. I checked my installed programs list, and sure enough, both versions appeared there. I uninstalled 2.4.0, and that's when the issue I described above started happening.

I tried uninstalling and reinstalling 2.4.1 to no avail, so I uninstalled it again and restarted. That's when something really, really weird happened: upon logging in, blobsaver started automatically as usual... _even though there was no longer any version of it installed on the computer._

I tried completely deleting any trace of the program (including the shortcut in the Startup folder), restarting and trying to reinstall it from a fresh download. I also tried using "Uninstall" in the Options menu. Neither worked; the program still hangs as soon as I hit "go," as does the debug log window if it's open.

**Desktop (please complete the following information):**
 - OS: Windows 10 version 1909
 - App version: 2.4.1

**Comment by airsquared** [comment]
Try [this](https://github.com/airsquared/blobsaver/files/4391055/blobsaver-windows.exe.zip) and send me the output displayed in the console window that opens

**Comment by skiasaurus** [comment]
````
path = C:\Program Files\blobsaver\lib
tsschecker = C:\Program Files\blobsaver\tsschecker.exe
````
That's it.

**Comment by airsquared** [comment]
That's all it shows even after attempting to save blobs?

**Comment by skiasaurus** [comment]
Correct. The second line doesn't show up until I hit "go."

**Comment by airsquared** [comment]
[This](https://github.com/airsquared/blobsaver/files/4391127/blobsaver-windows.exe.zip) should provide more output.

**Comment by skiasaurus** [comment]
Doesn't appear to.


![Annotation 2020-03-26 223052](https://user-images.githubusercontent.com/5809600/77715285-94872200-6fb1-11ea-84dd-a18bf43c033f.png)
![Annotation 2020-03-26 223052](https://user-images.githubusercontent.com/5809600/77715288-9650e580-6fb1-11ea-8ca1-15525c192c77.png)

**Comment by airsquared** [comment]
I've found the problem now. If you give it enough time, blobsaver will eventually unfreeze and successfully save blobs. It is currently taking so long because Apple changed the structure of their `.ipsw` files, and because of this blobsaver is attempting to download the whole ipsw file (~3.8 GB) to save blobs when it only needs the `BuildManifest.plist` file.

**Comment by airsquared** [comment]
I am currently trying to figure out another way to get `BuildManifest.plist` without needing to read the entire `.ipsw` file

**Comment by timidhermit** [comment]
airsquared, any update on this issue or a workaround in the meantime in order to user blobsaver v2.4.1?

I am trying to save SHSH2 blobs for iOS 13.4 and 13.4.1 for about half a dozen devices with presets. Each time I press GO, blobsaver completely locks up and becomes "non responsive". I can see from Network Connection meter that it is connecting online, presumably downloading the entire IPSW. But even after allowing one to finish (which took hours for each IPSW), the app stays "non responsive" and nothing happens. Please help!

**Comment by airsquared** [comment]
I've been able to fix this issue on macOS and Linux but I'm still working on a fix for Windows.

**Comment by timidhermit** [comment]
I am indeed using the Windows client. So are you saying the v2.4.1 of macOS and Linux is already fixed? Or are you referring to an upcoming build of blobsaver?

**Comment by airsquared** [comment]
Upcoming build

**Comment by airsquared** [comment]
@skiasaurus @timidhermit Please try [this](https://github.com/airsquared/blobsaver/files/4478825/blobsaver-windows.exe.zip), it should fix it.

**Comment by airsquared** [comment]
This bug has been fixed in v2.5.0.
